### PR TITLE
feat/Bonds

### PR DESCRIPTION
**Pull request by leonardishere**

### CODE REVIEW

File: contracts/BondDepository.sol [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    /**
+     * @notice Constructs the BondDepository contract.
+     * @param governance The address of the [governor](/docs/protocol/governance).
+     * @param solace Address of [**SOLACE**](./solace).
+     * @param solace Address of [**xSOLACE**](./xsolace).
+     * @param pool Address of [`UnderwritingPool`](./underwritingpool).
+     * @param solace Address of the DAO.
+     */
+    constructor(address governance, address solace, address xsolace, address pool, address dao) Governable(governance) {
```
Comment (ssozuer) [inline contracts/BondDepository.sol:35]: I think we can provide solace, xsolace, pool and dao via registry. But it is not big deal.

File: contracts/BondTellerErc20.sol [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+
+    /**
+     * @notice Calculate the current price of a bond.
+     * Assumes 1 SOLACE payout.
+     * @return price_ The price of the bond measured in `principal`.
+     */
+    function bondPrice() public view returns (uint256 price_) {
+        uint256 timeSinceLast = block.timestamp - lastPriceUpdate;
+        price_ = exponentialDecay(nextPrice, timeSinceLast);
+        if (price_ < minimumPrice) {
```
Comment (ssozuer) [inline contracts/BondTellerErc20.sol:134]: we did not set the default value for "minimumPrice" in the contract creation phase. Is it safe?

File: contracts/BondTellerErc20.sol [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+        // exchange rate
+        amountOut = 1 ether * amountIn / bondPrice(); // 1 ether => 1 solace
+        // ensure there is remaining capacity for bond
+        if (capacityIsPayout) {
+            // capacity in payout terms
+            require(capacity >= amountOut, "bond at capacity");
+        } else {
+            // capacity in principal terms
+            require(capacity >= amountIn, "bond at capacity");
+        }
```
Comment (ssozuer) [inline contracts/BondTellerErc20.sol:158]: require(capacityIsPayout && (capacity >= amountOut), "bond at capacity");
require(capacity >= amountIn, "bond at capacity");

Not big deal, just saving for the space.

File: contracts/CapitalPool.sol [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "./Governable.sol";
+import "./interface/ICapitalPool.sol";
+
+
+contract CapitalPool is ICapitalPool, ReentrancyGuard, Governable {
```
Comment (ssozuer) [inline contracts/CapitalPool.sol:12]: Missing contract doc.

File: contracts/BondDepository.sol (hunk 1 above)
Comment (leonardishere) [reply contracts/BondDepository.sol:35]: registry doesnt support those types. v2 should allow for arbitrary adds and lookups

File: contracts/BondTellerErc20.sol (hunk 2 above)
Comment (leonardishere) [reply contracts/BondTellerErc20.sol:134]: good catch, added initialization check

File: contracts/CapitalPool.sol (hunk 4 above)
Comment (leonardishere) [reply contracts/CapitalPool.sol:12]: added

**Pull request merged**